tlslite/messages.py

```diff
@@ -1765,3 +1765,74 @@ def parse(self, p):
 
     def write(self):
         return self.bytes
+
+
+class Heartbeat(object):
+    """
+    Handling Heartbeat extension from RFC 6520
+
+    @type message_type: int
+    @ivar message_type: type of message (response or request)
+
+    @type payload_length: int
+    @ivar payload_length: length of sending payload
+
+    @type payload: bytes
+    @ivar payload: payload
+    """
+    def __init__(self):
+        self.contentType = ContentType.heartbeat
+        self.message_type = 0
+        self.payload_length = 0
+        self.payload = bytearray(0)
+        self.padding = bytearray(0)
+        self.padding_length = 0
+
+    def create(self, message_type, payload_length, payload):
+        self.message_type = message_type
+        self.payload_length = payload_length
+        # dont check size, because of reproducing Heartbleed
+        self.payload = payload
+        self.padding_length = 2**14 - payload_length - 3
+        self.padding = bytearray(self.padding_length)
```

Comment:
the padding bytes need to be random if I'm reading the RFC correctly...

(just for clarity: I don't think the randomness should be inside `create()`, it's a general comment on the padding)